src/ekf.cpp

```diff
@@ -362,7 +362,26 @@ namespace RobotLocalization
     estimateErrorCovariance_ = (transferFunctionJacobian_ *
                                 estimateErrorCovariance_ *
                                 transferFunctionJacobian_.transpose());
-    estimateErrorCovariance_.noalias() += (processNoiseCovariance_ * delta);
+
+    // Compute processNoiseCovariance scaling if nessesary
+    Eigen::MatrixXd processNoiseCovariance = processNoiseCovariance_;
+    if (processNoiseCovarianceScaler_ > 0)
+    {
+      // Scale covariance in the direction of the velocity vector
+      Eigen::MatrixXd velScaler = state_.segment<6>(6).asDiagonal() * processNoiseCovarianceScaler_;
+
+      Eigen::Matrix3d R = (Eigen::AngleAxisd(state_(3), Eigen::Vector3d::UnitX())
+             * Eigen::AngleAxisd(state_(4), Eigen::Vector3d::UnitY())
+             * Eigen::AngleAxisd(state_(5), Eigen::Vector3d::UnitZ())).matrix();
+
+      Eigen::MatrixXd rotation = Eigen::MatrixXd::Identity(6,6);
+      rotation.block<3,3>(0,0) = R;
+      rotation.block<3,3>(3,3) = R;
+
+      velScaler  = rotation*velScaler*rotation.transpose();
+      processNoiseCovariance.block<6,6>(0,0) = velScaler*processNoiseCovariance_.block<6,6>(0,0)*velScaler.transpose();
```

Comment:
Not sure if it's fine to only touch the velocities and not the correlations wrt to the other state component, but I guess it should be fine :+1: